src/js_realm.hpp

```diff
@@ -579,13 +581,48 @@ void RealmClass<T>::wait_for_download_completion(ContextType ctx, FunctionType,
         });
         std::function<WaitHandler> waitFunc = std::move(wait_handler);
 
+        std::function<ProgressHandler> progressFunc; 
+
         auto realm = realm::Realm::get_shared_realm(config);
-        if (auto sync_config = config.sync_config) {
+        if (auto sync_config = config.sync_config)
+        {
+            static const String progressFuncName = "_onProgress";
+            bool progressFuncDefined = false;
+            if (!Value::is_boolean(ctx, sync_config_value) && !Value::is_undefined(ctx, sync_config_value))
```

Comment:
I took this check from populate sync config
https://github.com/realm/realm-js/blob/master/src/js_sync.hpp#L373-L375

It seems we support sync config as boolean to force sync on history